src/scala/org/pantsbuild/zinc/compiler/BUILD

```diff
@@ -12,6 +15,7 @@ scala_library(
     '3rdparty/jvm/com/martiansoftware:nailgun-server',
     '3rdparty/jvm/org/scala-lang/modules:scala-java8-compat',
     '3rdparty/jvm/org/scala-sbt:io',
+    '3rdparty/jvm/org/scala-sbt:sbt-compiler-bridge',
```

Comment:
Since we're planning to build the image dynamically, we should probably not add the compiler-bridge dep here... or we should add a separate target in this file (`zinc-for-native`) that you would use if you wanted to build the image?

So: yes to config being added to the main target, but no to adding the bridge?

Reply:
So I might be doing this wrong, but I believe the difference between this and the scala compiler jars (which are added to the command line which actually creates the native image) is that the sbt compiler bridge is *not* pinned to a specific scala patch version (in the way that e.g. `org.scala-lang:scala-library:2.12.3` is pinned). We also want to make sure that the compiler-bridge version is the same as the zinc version, which we get if we add it to the jar here, but would have to somehow ensure if we added it later. In terms of "where the classpath entries should be located", I think inserting this dep in here makes sense, as it's definitely going to be needed anyway at build time, and doesn't depend on the (specific) scala version we're compiling for.

Reply:
The zinc bootstrapper takes the sources of the jar, and then compiles them on a language-by-language basis: https://github.com/pantsbuild/pants/blob/e1b7fed8556a244d37e61a1d935a56d562be740a/src/scala/org/pantsbuild/zinc/bootstrapper/Cli.scala#L38-L45

I think the effect of what you're doing here is to embed a pre-compiled version... but it will be specific to whatever version of scala you are currently using.

Reply:
Ok. I'll have to add a method for additional classpath entries at build time just to inject the scala compiler/library/etc, so there's no reason that wouldn't be able to contain the compiler-bridge jar.

Reply:
> but it will be specific to whatever version of scala you are currently using.

But also, how does coursier/etc resolve this then, if the jars are only marked with `_2.12`, and the version (1.1.7) isn't related to the scala version? I don't understand what information coursier has that could let it be attached to a single scalac version.